server/signedcontainer/verify.go

```diff
@@ -0,0 +1,130 @@
+package signedcontainer
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/GoogleCloudPlatform/confidential-space/server/signedcontainer/internal/convert"
+	"github.com/tink-crypto/tink-go/v2/keyset"
+	tinksig "github.com/tink-crypto/tink-go/v2/signature"
+)
+
+type ImageSignature struct {
+	Payload   []byte
+	Signature []byte
+	Alg       SigningAlgorithm
+}
+
+const maxSignatureCount = 300
+
+// Verify attempts to verify the provided signatures with imageDigest and returns:
+//   - []*ImageSignature - list of successfully verified signatures
+//   - []error           - (nonblocking) list of errors from unsuccessful verifications
+//   - err               - (blocking) whether the number of provided signatures exceeds the allowed max
+func Verify(imageDigest string, signatures []*ImageSignature) ([]*ImageSignature, []error, error) {
```

Comment:
Why not accept a slice of `oci.Signature` as input? Since oci.Signature already has fields necessary for validation on the service side, we can convert it at agent level before sending to service (ITA/GCA), and the service can leverage this open source `Verify` for signature verification.

Reply:
If the agent converts it away from an oci.Signature in order to send it to the service, wouldn't we want to accept the end result of that conversion instead?

Reply:
do you mean the `ImageSignature` you defined here is "the end result of that conversion"? My view of the end result is some format that conforms to the service protocol. e.g., for GCA, the end result is the proto buffer `confidentialcomputingpb.ContainerSignature`. I'm unaware of what format of container signature ITA client accepts, but should be something similar.

Reply:
> do you mean the ImageSignature you defined here is "the end result of that conversion"
That's what I had in mind since it reflects what the GCA inputs to `verifyContainerImageSignatures`. Hence why I don't think `[]oci.Signature` input makes sense because the agent should have already converted by the time it reaches the verification service using this method.

Reply:
> > do you mean the ImageSignature you defined here is "the end result of that conversion"
> > That's what I had in mind since it reflects what the GCA inputs to `verifyContainerImageSignatures`. Hence why I don't think `[]oci.Signature` input makes sense because the agent should have already converted by the time it reaches the verification service using this method.

now i see your idea. If we view `ImageSignature` as an adapter between `oci.Signature` and service inputs, then it makes sense to abstract out this layer.